File: .sling/libs/web5/eqela/jk-data/0.32.0/jk.slon.convert.js
let JkSlonConvertToJson = function() {};

JkSlonConvertToJson.prototype._t = { "JkSlonConvertToJson" : true };
JkSlonConvertToJson.prototype._tobj = JkSlonConvertToJson;

JkSlonConvertToJson.NEW = function() {
	var v = new JkSlonConvertToJson;
	return v.CTOR_JkSlonConvertToJson();
};

JkSlonConvertToJson.prototype.CTOR_JkSlonConvertToJson = function() {
	return this;
};

JkSlonConvertToJson._main = function(args) {
	var src = args[0];
	if(JkLangString.isEmpty(src)) {
		return 1;
	}
	var dst = args[1];
	try {
		var data = JkSlonSLONParser.forFile((JkFsFile.forPath(src))).acceptObject();
		if(!(data != null)) {
			return 1;
		}
		var dstr = JkJsonJSONEncoder.toString(data);
		if(JkLangString.isNotEmpty(dst)) {
			var dstf = JkFsFile.forPath(dst);
			if(!dstf.setContentsUTF8(dstr)) {
				JkLangError._throw("failedToWrite", dstf);
			}
		}
		else {
			console.log(dstr);
		}
		return 0;
	}
	catch(e) {
		console.log("ERROR: " + (JkLangString.safeString((JkLangString.asString(e)))));
	}
	return 1;
};

JkSlonConvertToJson._main();

JkSlonConvertToJson.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkSlonConvertToJson"] === true;
};

File: .sling/libs/sushi/eqela/jk-data/0.23.0/jk.slon.convert.lua
jk = jk or {}

jk.slon = jk.slon or {}

jk.slon.convert = jk.slon.convert or {}

jk.slon.convert.ToJson = {}
jk.slon.convert.ToJson.__index = jk.slon.convert.ToJson
_vm:set_metatable(jk.slon.convert.ToJson, {})

function jk.slon.convert.ToJson._create()
	local v = _vm:set_metatable({}, jk.slon.convert.ToJson)
	return v
end

function jk.slon.convert.ToJson:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.slon.convert.ToJson'
	self['_isType.jk.slon.convert.ToJson'] = true
end

function jk.slon.convert.ToJson:_construct0()
	jk.slon.convert.ToJson._init(self)
	return self
end

function jk.slon.convert.ToJson:_main(args)
	local src = args[0 + 1]
	if _g.jk.lang.String:isEmpty(src) then
		do return 1 end
	end
	do
		local dst = args[1 + 1]
		do
			local _rval = nil
			local _status, _err = _vm:execute_protected_call(function()
				local data = _g.jk.slon.SLONParser:forFile(_g.jk.fs.File:forPath(src)):acceptObject()
				if not (data ~= nil) then
					_rval = {}
					_rval.value = 1
					do return end
				end
				do
					local dstr = _g.jk.json.JSONEncoder:toString(data)
					if _g.jk.lang.String:isNotEmpty(dst) then
						local dstf = _g.jk.fs.File:forPath(dst)
						if not dstf:setContentsUTF8(dstr) then
							do _g.jk.lang.Error:throw("failedToWrite", dstf) end
						end
					else
						_io:write_to_stdout(dstr .. "\n")
					end
					_rval = {}
					_rval.value = 0
					do return end
				end
			end)
			if _err then
				local e = _err
				_io:write_to_stdout("ERROR: " .. _g.jk.lang.String:safeString(_g.jk.lang.String:asString(e)) .. "\n")
			end
			if _rval then
				do return _rval.value end
			end
		end
		do return 1 end
	end
end

function _main(args)
	do return jk.slon.convert.ToJson:_main(args) end
end
